src/NuGet.Core/NuGet.ProjectModel/LockFile/Utf8JsonStreamAssetsLogMessageConverter.cs

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using NuGet.Common;
+
+namespace NuGet.ProjectModel
+{
+    /// <summary>
+    /// A <see cref="Utf8JsonStreamReaderConverter{T}"/> to allow read JSON into <see cref="AssetsLogMessage"/>
+    /// </summary>
+    internal class Utf8JsonStreamAssetsLogMessageConverter : Utf8JsonStreamReaderConverter<AssetsLogMessage>
+    {
+        private static readonly byte[] Utf8Level = Encoding.UTF8.GetBytes(LogMessageProperties.LEVEL);
+        private static readonly byte[] Utf8Code = Encoding.UTF8.GetBytes(LogMessageProperties.CODE);
+        private static readonly byte[] Utf8WarningLevel = Encoding.UTF8.GetBytes(LogMessageProperties.WARNING_LEVEL);
+        private static readonly byte[] Utf8FilePath = Encoding.UTF8.GetBytes(LogMessageProperties.FILE_PATH);
+        private static readonly byte[] Utf8StartLineNumber = Encoding.UTF8.GetBytes(LogMessageProperties.START_LINE_NUMBER);
+        private static readonly byte[] Utf8StartColumnNumber = Encoding.UTF8.GetBytes(LogMessageProperties.START_COLUMN_NUMBER);
+        private static readonly byte[] Utf8EndLineNumber = Encoding.UTF8.GetBytes(LogMessageProperties.END_LINE_NUMBER);
+        private static readonly byte[] Utf8EndColumnNumber = Encoding.UTF8.GetBytes(LogMessageProperties.END_COLUMN_NUMBER);
+        private static readonly byte[] Utf8Message = Encoding.UTF8.GetBytes(LogMessageProperties.MESSAGE);
+        private static readonly byte[] Utf8LibraryId = Encoding.UTF8.GetBytes(LogMessageProperties.LIBRARY_ID);
+        private static readonly byte[] Utf8TargetGraphs = Encoding.UTF8.GetBytes(LogMessageProperties.TARGET_GRAPHS);
+
+        public override AssetsLogMessage Read(ref Utf8JsonStreamReader reader)
+        {
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                throw new JsonException("Expected StartObject, found " + reader.TokenType);
+            }
+
+            var isValid = true;
+            LogLevel level = default;
+            NuGetLogCode code = default;
+            //matching default warning level when AssetLogMessage object is created
+            WarningLevel warningLevel = WarningLevel.Severe;
+            string message = default;
+            string filePath = default;
+            int startLineNumber = default;
+            int startColNumber = default;
+            int endLineNumber = default;
+            int endColNumber = default;
+            string libraryId = default;
+            IReadOnlyList<string> targetGraphs = null;
+
+            while (reader.Read() && reader.TokenType == JsonTokenType.PropertyName)
+            {
+                if (!isValid)
+                {
+                    reader.TrySkip();
```

Comment:
What does it mean if `TrySkip` returns false? Do we need to handle it?

Maybe this is a copy/port from the existing NJ code, but if it's not valid, why don't we immediately throw an exception? NuGet writes the assets file, we don't expect 3rd parties to create this file, so it's not obvious to me why would try to "gracefully" handle errors.